scripts/api-compare/api-compare.mjs

```diff
@@ -0,0 +1,350 @@
+
+		if (!fs.existsSync(outputFileDirPath)) {
+			fs.mkdirSync(outputFileDirPath)
+		}
+
+		const apiURLFileName = newApiURL.replace(/\//g, '_').replace(/:/g, '-')
+		const outputFile = path.join(outputFileDirPath, `${apiURLFileName}.txt`)
+
+		const strippedOutputString = stripAnsi(outputString)
+		fs.writeFileSync(outputFile, strippedOutputString)
+	}
+}
+
+async function fetchApiTypeVersionAndNav(options, product, versionMetadata) {
+	let newApiURL
+	let oldApiURL
+
+	if (options.api === 'version-metadata') {
+		newApiURL = `${options.newApiUrl}/api/content/${product}/version-metadata`
+		oldApiURL = `${options.oldApiUrl}/api/content/${product}/version-metadata?partial=true`
+	} else if (options.api === 'nav-data') {
+		newApiURL = `${options.newApiUrl}/api/content/${product}/nav-data/${versionMetadata.version}`
+		oldApiURL = `${options.oldApiUrl}/api/content/${product}/nav-data/${versionMetadata.version}`
+	}
+
+	const newApiResponse = await fetch(newApiURL)
+	const oldApiResponse = await fetch(oldApiURL)
+
+	const newApiData = await newApiResponse.json()
+	const oldApiData = await oldApiResponse.json()
+
+	const newApiDataStrings = JSON.stringify(
+		sortObjectByKeys(newApiData.result),
+		null,
+		2
+	).split('\n')
+	const oldApiDataStrings = JSON.stringify(
+		sortObjectByKeys(oldApiData.result),
+		null,
+		2
+	).split('\n')
+
+	return { newApiDataStrings, oldApiDataStrings, newApiURL }
+}
+
+const testsPassed = []
+const testsFailed = []
+for (const [product, versions] of Object.entries(versionMetadata)) {
+	if (options.product && options.product !== product) {
+		continue
+	}
+
+	for (const versionMetadata of versions) {
+		if (options.version && options.version !== versionMetadata.version) {
+			continue
+		}
+
+		if (options.api === 'version-metadata' || options.api === 'nav-data') {
+			const { newApiDataStrings, oldApiDataStrings, newApiURL } =
+				await fetchApiTypeVersionAndNav(options, product, versionMetadata)
+
+			const diffOptions = {
+				contextLines: 1,
+				expand: false,
+			}
+
+			const difference = diffLinesUnified(
+				oldApiDataStrings,
+				newApiDataStrings,
+				diffOptions
+			)
+
+			const outputString = `Testing API URL:\n${newApiURL}\n${difference}`
+			console.log(outputString)
+
+			saveTestOutputIfSelected(outputString, newApiURL)
+			break
+		} else if (options.api === 'content') {
+			const productContentDir = contentDirMap[product]
+			const contentPath = path.join(
+				mainContentDirectory,
+				product,
+				versionMetadata.version,
+				productContentDir
+			)
+
+			if (!fs.existsSync(contentPath)) {
+				console.log(`Directory ${contentPath} does not exist`)
+				continue
+			}
+
+			const apiPaths = getAllContentApiPaths(contentPath)
+
+			const randomIndexes = []
+			const numOfRandomIndexes =
+				Math.min(options.numOfTests, apiPaths.length) || 1
+
+			// let apiPathsLeft = [...apiPaths];
+			while (randomIndexes.length < numOfRandomIndexes) {
```

Comment:
This is the "fuzzy" part where it randomly tests `/doc` routes, I don't think this is necessary anymore.